### PR TITLE
Check tool call type instead of function field for tool call data support

### PR DESCRIPTION
**Pull request by rossdanlm**
Check tool call type instead of function field for tool call data support

Just a small nit I noticed from Sarmad's changes in https://github.com/lastmile-ai/aiconfig/pull/752. I also ran the autoformatter command: `fd --glob '*.py'  python/src/aiconfig/default_parsers | xargs python -m 'scripts.lint' --mode=fix --files`.

I also updated the 3 ipynb notebooks and found + fixed some bugs from that! I also cleared the outputs here, not sure if that's desired, but tested that it works so should be good!

## Test Plan
Run the anyscale, openai, and function call cookbooks to make sure they still work

Screenshot 2024-01-04 at 22 13 11
Screenshot 2024-01-04 at 22 13 14
Screenshot 2024-01-04 at 22 35 08

### CODE REVIEW

File: cookbooks/Function-Calling-OpenAI/python/Function_Calling_with_OpenAI.ipynb [hunk 1]
```diff
@@ -267,10 +267,9 @@
         "params = {\"book\":\"Where the Crawdads Sing\"}\n",
         "inference_options = InferenceOptions(stream=True)\n",
         "\n",
-        "completion = await config.run(\"recommend_book\", params, options=inference_options)\n",
-        "\n",
-        "func_call = completion[0].data.get('function_call')\n",
-        "print(func_call)\n"
+        "result = await config.run(\"recommend_book\", params, options=inference_options)\n",
+        "tool_calls = result[0].data.value\n",
+        "func_call = tool_calls[0].function\n"
```
Comment (rossdanlm) [inline cookbooks/Function-Calling-OpenAI/python/Function_Calling_with_OpenAI.ipynb:272]: These changes actually fix something that was broken in the cookbook

**Review by saqadri — COMMENTED** [review]
Please also update anyscale_endpoint.py with the same changes as in OpenAI.py

**Review by saqadri — APPROVED** [review]
Looks good, just the one comment to update anyscale_endpoint.py with the same changes as in OpenAI.py

**Comment by rossdanlm** [comment]
> Please also update anyscale_endpoint.py with the same changes as in OpenAI.py

This is actually already done because anyscale inherits from OpenAI:

> class AnyscaleEndpoint(OpenAIInference):

We can probably actually just make `build_output_data` a class method inside of `OpenAIInference` file so that we dont' have to redo two callsites on changes. Made bootcamp task for it in https://github.com/lastmile-ai/aiconfig/issues/767

**Pull request merged**